### DISCUSSION TITLE
introduce useContractDeploy hook

### DISCUSSION THREAD

**Discussion by enochakinbode** [wevm/wagmi#1950] [Ideas]
Hi,
I recently integrated wagmi into a project of mine it was a great experience. I needed a way to deploy a contract from the frontend. I was able to get this working using some exposed utility functions and hooks, but I feel it will be nice if a native hook, `useContractDeploy`, can be introduced for this purpose. I'd also love to contribute to its development. 

Here is an minor implementation I tweaked for my own use.
[Hook](https://github.com/iamenochlee/dalle2nft/blob/master/frontend/hooks/useContractDeploy.tsx).

Hoping to here back from you, Thanks.

**Comment by jcislo** [comment]
Hey are there any updates on this?

**Comment by NickStein85** [comment]
It would be awesome if we can deploy contracts via wagmi

**Comment by ameya-deshmukh** [comment]
This is very useful, thank you! And I agree, this would be a very nice feature :)
